### PR TITLE
drivers: imx: edma: translate burst size in bytes

### PR DESCRIPTION
**Pull request by groncarolo**
burst_size represents the FIFO width in words, translate it
into bytes to have homogeneous quantities

Signed-off-by: Guido Roncarolo <guido.roncarolo@nxp.com>

### CODE REVIEW

**Comment by paulstelian97** [comment]
Sorry, cannot review. I need to understand the burst_elems field itself and properly use it first. Do you have any explanation on how that works?

File: src/drivers/imx/edma.c [hunk 1]
```diff
@@ -497,19 +503,23 @@ static int edma_get_data_size(struct dma_chan_data *channel,
...
 	 * interrupt.
 	 */
+	int32_t data_size;
+
+	data_size = (int32_t)dma_chan_reg_read(channel, EDMA_TCD_SLAST);
+
 	switch (channel->direction) {
 	case SOF_IPC_STREAM_PLAYBACK:
-		*free = dma_chan_reg_read(channel, EDMA_TCD_NBYTES);
+		*free = ABS(data_size) / 2;
```
Comment (paulstelian97) [inline src/drivers/imx/edma.c]: Too many things in the same commit, I think this should be separate.

Comment (paulstelian97) [reply src/drivers/imx/edma.c]: Disregard this, didn't notice they _were_ separate.

**Comment by groncarolo** [comment]
no problem

we should transfer in general half of the fifo size maximum each time, i.e so 32 words for SAI.
the number of DMA transfers for one half major loop will be 3.

We use the FIFO size to determine what is the most convenient size to use as
number of bytes transferred by each DMA transfer

In particular the FIFO size is in words and have to be calculated in bytes, the we take half since we
don't want to push more data than that

this means that in the ESAI case the FIFO 128 words the element size is 384
we will have a size of 256
total size is 768
BITER and BITER is going to be 3

**Comment by paulstelian97** [comment]
CITER and BITER being 3 means that we will get interrupts irregularly, namely after the first 512 bytes and after the remaining 256. This will probably cause more mayhem if we're not very careful.

**Comment by paulstelian97** [comment]
Without this PR I didn't hear sound well at 256, only at 384 for depth on the ESAI. I think if I set it to 512 I would overrun the FIFO most likely.

So I think this depends on the actual way the DAI is configured (right now I'm hardcoding the watermarks, and this .depth field MUST depend on what the watermarks are set as)

**Comment by groncarolo** [comment]
you are correct the FIFO watermark is linked to the chunk size that is calculated here
the matching value here is `FIFO size / 2 `

**Comment by paulstelian97** [comment]
So after this is merged I'd set it to 192 (which fits my current value for the watermark). Got it. Although the comments/description only become more confusing due to this.

**Comment by jlacla** [comment]
I agree with this commit, even if as co-author I am not sure to be morally eligible for a review of it.

**Comment by jlacla** [comment]
I just found a small error in the Guido's comment. Fixing it for clarity.

> this means that in the ESAI case the FIFO 128 words the element size is 384
> we will have a size of 256
> total size is 768
> BITER and BITER is going to be 3     <= error there

Each buffer is 384 bytes = 96 words => 3 DMA transfers of 32 words for each buffer 
=> BITER and CITER are equal to 6 because we have two buffers. The half interrupt occurs when CITER=3.

**Comment by paulstelian97** [comment]
> I just found a small error in the Guido's comment. Fixing it for clarity.
> 
> > this means that in the ESAI case the FIFO 128 words the element size is 384
> > we will have a size of 256
> > total size is 768
> > BITER and BITER is going to be 3     <= error there
> 
> Each buffer is 384 bytes = 96 words => 3 DMA transfers of 32 words for each buffer
> => BITER and CITER are equal to 6 because we have two buffers. The half interrupt occurs when CITER=3.

Oh, that was actually what was unclear to me. Having an odd number of minor loops would make the INTHALF interrupt behave in a way that doesn't play well with SOF without extra changes.

**Comment by jlacla** [comment]
Yes, due to the fact we have two buffers, the value of BITER is always even.

**Comment by paulstelian97** [comment]
Ah perfect, SOF wants to know when each of the buffers is finished. You already have my approve, guess @dbaluta can now merge it and I can update the ESAI driver.

**Comment by paulstelian97** [comment]
SOFCI TEST

**Pull request merged**